File: app/src/main/java/com/dhu/dhu_timetable/repo/NoticeRepo.java
package com.dhu.dhu_timetable.repo;

import android.util.Log;

import androidx.lifecycle.LiveData;

import com.dhu.dhu_timetable.ui.navitem.notice.NoticeModel;
import com.dhu.dhu_timetable.network.NoticeApiClient;

import java.util.List;

public class NoticeRepo {
    private NoticeApiClient mNoticeApiClient;

    // 싱글톤 패턴
    private static NoticeRepo instance;

    public static NoticeRepo getInstance() {
        if (instance == null)
            instance = new NoticeRepo();
        return instance;
    }

    public NoticeRepo() {
        mNoticeApiClient = NoticeApiClient.getInstance();
        Log.d("jaemin", "NoticeRepo: 성공");
    }

    public LiveData<List<NoticeModel>> getNoticeData(){
        return mNoticeApiClient.getNoticeData();
    }

    public void setNoticeData(){
        mNoticeApiClient.setNoticeData();
    }

}

File: app/src/main/java/com/dhu/dhu_timetable/repo/NoticeRepo.kt
package com.dhu.dhu_timetable.repo

import androidx.lifecycle.LiveData
import com.dhu.dhu_timetable.model.NoticeModel
import com.dhu.dhu_timetable.network.NoticeApiClient

class NoticeRepo {
    private val mNoticeApiClient: NoticeApiClient by lazy { NoticeApiClient.getInstance() }

    companion object {
        @Volatile private var instance: NoticeRepo? = null
        @JvmStatic
        fun getInstance(): NoticeRepo =
                instance ?: synchronized(this) {
                    instance ?: NoticeRepo().also { instance = it }
                }
    }


    fun getNoticeData(): LiveData<List<NoticeModel>> {
        return mNoticeApiClient.noticeData
    }

    fun setNoticeData() {
        mNoticeApiClient.setNoticeData()
    }
}